src/systems/diffeqs/odesystem.jl

```diff
@@ -92,6 +92,31 @@ struct ODESystem <: AbstractODESystem
     end
 end
 
+function move_diffs(eq::Equation,r)
+    # Do not modify `D(x) ~ ...`, already correct
+    # Ignore `δ(x) ~ ...` for now
+    if !(eq.lhs isa Term && ModelingToolkit.operation(eq.lhs) isa Differential) &&
+       !(eq.lhs isa Term && ModelingToolkit.operation(eq.lhs) isa Difference)
+       _eq = eq.rhs-eq.lhs
+       rhs = r(_eq)
+       if isnothing(rhs)
```

Comment:
```suggestion
       if rhs === nothing
```
`isnothing` probably should never be used, and just removed from Base.